sql/select_test.go

```diff
@@ -308,6 +308,11 @@ func TestApplyConstraints(t *testing.T) {
 		{`(a, b) = (1, 2) AND c IS NOT NULL`, []string{"a", "b", "c"}, `<nil>`},
 		{`a IN (1, 2) AND b = 3`, []string{"a", "b"}, `b = 3`},
 		{`a <= 5 AND b >= 6 AND (a, b) IN ((1, 2))`, []string{"a", "b"}, `a <= 5 AND b >= 6`},
+		{`a IN (1) AND a = 1`, []string{"a"}, `<nil>`},
+		// Filters that are not trimmed as of Dec 2015, although they could be.
+		// Issue #3473.
+		// {`a > 1`, []string{"a"}, `<nil>`},
+		// {`a < 1`, []string{"a"}, `<nil>`},
```

Comment:
I seem to recall some issue with `NULL` and trimming of filters, but the exact scenario is escaping me right now.